athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/propertygraph/PropertyGraphHandler.java

```diff
@@ -0,0 +1,231 @@
+ * this class for each 'Split' you generated in NeptuneMetadataHandler.
+ * <p>
+ * For more examples, please see the other connectors in this repository (e.g.
+ * athena-cloudwatch, athena-docdb, etc...)
+ */
+public class PropertyGraphHandler 
+{
+    private static final Logger logger = LoggerFactory.getLogger(NeptuneRecordHandler.class);
+
+    /**
+     * used to aid in debugging. Athena will use this name in conjunction with your
+     * catalog id to correlate relevant query errors.
+     */
+
+    private final NeptuneConnection neptuneConnection;
+
+    @VisibleForTesting
+    public PropertyGraphHandler(NeptuneConnection neptuneConnection) 
+    {
+        this.neptuneConnection = neptuneConnection;
+    }
+
+    /**
+     * Used to read the row data associated with the provided Split.
+     *
+     * @param spiller            A BlockSpiller that should be used to write the row
+     *                           data associated with this Split. The BlockSpiller
+     *                           automatically handles chunking the response,
+     *                           encrypting, and spilling to S3.
+     * @param recordsRequest     Details of the read request, including: 1. The
+     *                           Split 2. The Catalog, Database, and Table the read
+     *                           request is for. 3. The filtering predicate (if any)
+     *                           4. The columns required for projection.
+     * @param queryStatusChecker A QueryStatusChecker that you can use to stop doing
+     *                           work for a query that has already terminated
+     * @throws Exception
+     * @note Avoid writing >10 rows per-call to BlockSpiller.writeRow(...) because
+     *       this will limit the BlockSpiller's ability to control Block size. The
+     *       resulting increase in Block size may cause failures and reduced
+     *       performance.
+     */
+
+    public void executeQuery(ReadRecordsRequest recordsRequest, final QueryStatusChecker queryStatusChecker,
+            final BlockSpiller spiller) throws Exception 
+    {
+        logger.info("readWithConstraint: enter - " + recordsRequest.getSplit());
+        long numRows = 0;
+        Client client = neptuneConnection.getNeptuneClientConnection();
+        GraphTraversalSource graphTraversalSource = neptuneConnection.getTraversalSource(client);
+        GraphTraversal graphTraversal = null;
+        TableName tableName = recordsRequest.getTableName();
+        String labelName = tableName.getTableName();
+        String tableSchemaMetaType = recordsRequest.getSchema().getCustomMetadata().get("type");
+        GeneratedRowWriter.RowWriterBuilder builder = GeneratedRowWriter.newBuilder(recordsRequest.getConstraints());
+        
+        logger.info("readWithConstraint: schema type is " + tableSchemaMetaType);
+
+        if (tableSchemaMetaType != null) {
+            switch (tableSchemaMetaType) {
+                case "vertex":
```

Comment:
can we make this an Enum?

Reply:
create a file Enums.java for all enums: https://github.com/abhishekpradeepmishra/aws-athena-query-federation/blob/athena-neptune-edgesupport/athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/propertygraph/Enums.java